core/lib/types/src/commitment/mod.rs

```diff
@@ -670,6 +729,8 @@ pub enum CommitmentInput {
         state_diffs: Vec<StateDiffRecord>,
         aux_commitments: AuxCommitments,
         blob_commitments: Vec<H256>,
+        blob_linear_hashes: Vec<H256>,
```

Comment:
Could make sense to define `blob_commitments_and_hashes: Vec<(H256, H256)>` to make (some) invalid states non-representable. Or even define a struct with `commitment` and `linear_hash` fields.